lib/hammer_cli_foreman/attribute.rb

```diff
@@ -0,0 +1,282 @@
+      params
+    end
+
+    class Create < HammerCLIForeman::CreateCommand
+        desc _('Create compute profile set of values.')
+
+      option '--compute-attributes', 'COMPUTE_ATTRS', _('Compute resource attributes'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--interface', 'INTERFACE', _('Interface parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new, :multivalued => true
+      option '--volume', 'VOLUME', _('Volume parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new, :multivalued => true
+
+      extend_help do |h|
+        ::HammerCLIForeman::ComputeResources.extend_help(h, :all)
+      end
+
+      validate_options do
+        any(:option_compute_profile_id, :option_compute_profile_name ).required
+        any(:option_compute_resource_id, :option_compute_resource_name).required
+      end
+
+      def request_params
+        params = super
+        params['compute_attribute']['vm_attrs'] = option_compute_attributes || {}
+        if option_interface
+          params['compute_attribute']['vm_attrs']['interfaces_attributes'] = {new_element_id => option_interface}
+        end
+        if option_volume
+            params['compute_attribute']['vm_attrs']['volumes_attributes'] = {new_element_id => option_volume}
+        end
+        params
+      end
+
+      success_message _('Compute profile attributes are set.')
+      failure_message _('Could not set the compute profile attributes')
+      build_options
+    end
+
+    class Update < HammerCLIForeman::UpdateCommand
+      desc _('Update compute profile values.')
+
+      option '--compute-attributes', 'COMPUTE_ATTRS', _('Compute resource attributes, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--interface', 'INTERFACE', _('Interface parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new, :multivalued => true
+      option '--volume', 'VOLUME', _('Volume parameters, should be comma separated list of values'),
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new, :multivalued => true
+
+      extend_help do |h|
+        ::HammerCLIForeman::ComputeResources.extend_help(h, :all)
+      end
+
+      validate_options do
+        any(:option_compute_profile_id, :option_compute_profile_name ).required
+        any(:option_compute_resource_id, :option_compute_resource_name).required
+        any(:option_interface, :option_volume, :option_compute_attributes).required
+      end
+
+      def request_params
+        params = HammerCLIForeman::Attribute.get_params(options)
```

Comment:
@shiramax, while testing the update it didn't work as expected. E.g. 
```
hammer  compute-profile values update --compute-profile-id 1 --compute-resource-id 1 --compute-attributes cpus=8 --volume capacity=8G
``` 
after fixing some issues caused I've lost interfaces and volumes attributes. The `option_compute_attributes` replaced all `vm_attrs` including the volumes and interfaces. The volumes then failed as it was sent as a list. I'd expect the example command will replace compute_attributes, volume_attributes and keep existing interface attributes. I also found out the 'interfaces_attributes` should be called `nics_attributes`. Not sure if both are accepted but nics_attributes worked for me.

Overall patch to the `request_params` that worked for me would be:
```diff
diff --git a/lib/hammer_cli_foreman/attribute.rb b/lib/hammer_cli_foreman/attribute.rb
index 0fd4e60..525cecf 100644
--- a/lib/hammer_cli_foreman/attribute.rb
+++ b/lib/hammer_cli_foreman/attribute.rb
@@ -15,8 +15,8 @@ module HammerCLIForeman
       profile = HammerCLIForeman.record_to_common_format(
           HammerCLIForeman.foreman_resource(:compute_profiles).call(:show, 'id' => options['option_compute_profile_id'] )
       )
-      HammerCLIForeman.foreman_resource(:compute_resources).call(:show, 'id' => options['option_compute_resource_id'] )
       params['compute_attribute'] = profile['compute_attributes'].select { |hash| hash['compute_resource_id'] == options['option_compute_resource_id']}[0] || {}
+      params["compute_attribute"].delete("attributes")
       params
     end
 
@@ -73,22 +73,35 @@ module HammerCLIForeman
       validate_options do
         any(:option_compute_profile_id, :option_compute_profile_name ).required
         any(:option_compute_resource_id, :option_compute_resource_name).required
-        any(:option_interface, :option_volume, :option_compute_attributes).required
       end
 
       def request_params
         params = HammerCLIForeman::Attribute.get_params(options)
+        vm_attrs = params['compute_attribute']['vm_attrs']
+        original_volumes = vm_attrs['volumes_attributes'] || {}
+        original_interfaces = vm_attrs['nics_attributes'] || {}
         params['id'] =  params['compute_attribute']['id']
-        params['compute_attribute']['vm_attrs'] = option_compute_attributes if option_compute_attributes
-        params['compute_attribute']['vm_attrs']['interfaces_attributes'] = option_interface if option_interface
-        params['compute_attribute']['vm_attrs']['volumes_attributes'] = option_compute_attributes if option_volume
+        if option_compute_attributes
+          vm_attrs = option_compute_attributes
+          vm_attrs['volumes_attributes'] ||= original_volumes
+          vm_attrs['nics_attributes'] ||= original_interfaces
+        end
+        vm_attrs['nics_attributes'] = attribute_hash(option_interface_list) unless option_interface_list.empty?
+        vm_attrs['volumes_attributes'] = attribute_hash(option_volume_list) unless option_volume_list.empty?
+        params['compute_attribute']['vm_attrs'] = vm_attrs
         params
       end
+
+      def attribute_hash(attribute_list)
+        attribute_list.size.times.map { |idx| idx.to_s }.zip(attribute_list).to_h
+      end
+
       success_message _('Compute profile attributes updated.')
       failure_message _('Could not update the compute profile attributes')
 
       build_options :without => :id
     end
+
     # Using the Update command because adding a new interface is done by modifying existing compute_attribute
     class AddInterface < HammerCLIForeman::UpdateCommand
       command_name 'add-interface'
```
Could you check if it matches your expectations and confirm it works for you?
If so I think we will need to update also the other create/update commands. Let me know if you need some help.